CMakeLists.txt

```
@@ -299,21 +300,33 @@ endif()
 if( TRACCC_BUILD_ALPAKA )
    add_subdirectory( device/alpaka )
 endif()
-add_subdirectory( io )
-add_subdirectory( performance )
+if ( TRACCC_BUILD_IO )
+   add_subdirectory( io )
+   add_subdirectory( performance )
+   add_subdirectory( simulation )
+else()
+   message(STATUS "traccc::io not built, traccc::performance and traccc::simulation are forcefully switched off.")
+endif()
+
 add_subdirectory( plugins )
-add_subdirectory( simulation )
 
 if ( TRACCC_BUILD_EXAMPLES )
    # Find Boost.
    find_package( Boost REQUIRED COMPONENTS program_options filesystem)
-
-   add_subdirectory( examples )
+   if ( TRACCC_BUILD_IO )
+      add_subdirectory( examples )
+   else()
+      message(STATUS "traccc::io not built, traccc::examples are forcefully switched off.")
+   endif()
```

I'd prefer if this was a bit more vocal and explicit, i.e. I don't think it's a good idea for there to be any possibility for the user to specify `TRACCC_BUILD_EXAMPLES` and then for the build system to decide not to do it. I would suggest the following:

```suggestion
   if ( NOT TRACCC_BUILD_IO )
      message(FATAL_ERROR "traccc::io is disabled, but it is required to build the examples.")
   endif()
   add_subdirectory( examples )
```

---

You have a good point here!